lib/livebook_web/live/hub/teams/deployment_group_instance_component.ex

```diff
@@ -0,0 +1,249 @@
+defmodule LivebookWeb.Hub.Teams.DeploymentGroupInstanceComponent do
+  use LivebookWeb, :live_component
+
+  alias Livebook.Hubs
+
+  @impl true
+  def mount(socket) do
+    {:ok, assign(socket, messages: [])}
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    socket = assign(socket, assigns)
+
+    socket =
+      if socket.assigns[:agent_key_id] do
+        socket
+      else
+        agent_key_id =
+          case assigns.deployment_group.agent_keys do
+            [%{id: id} | _] -> id
+            _ -> nil
+          end
+
+        assign(socket, :agent_key_id, agent_key_id)
+      end
+
+    socket =
+      assign_new(socket, :changeset, fn ->
+        Hubs.Dockerfile.config_changeset(base_config(socket))
+      end)
+
+    {:ok, update_instructions(socket)}
+  end
+
+  defp base_config(socket) do
+    Hubs.Dockerfile.from_deployment_group(socket.assigns.deployment_group)
+  end
+
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div class="p-6 max-w-4xl flex flex-col space-y-3">
+      <h3 class="text-2xl font-semibold text-gray-800">
+        App server setup
+      </h3>
+
+      <p class="text-gray-700">
+        App server is an instance of Livebook where you can deploy Livebook
+        apps via Livebook Teams. Each app server belongs to a specific
+        deployment group.
+      </p>
+
+      <.content
+        hub={@hub}
+        deployment_group={@deployment_group}
+        myself={@myself}
+        agent_key_id={@agent_key_id}
+        changeset={@changeset}
+        instructions={@instructions}
+        messages={@messages}
+      />
+    </div>
+    """
+  end
+
+  defp content(%{deployment_group: %{agent_keys: []}} = assigns) do
```

Comment:
We have a feature to ensure there is at least one agent key, so I think we can remove this one. :)

Reply:
We create one by default but it can still be removed manually. Should I add a validation to ensure at least one on the Teams side then?

Reply:
There is an open issue to do so, so it will be done sooner than later :D

Reply:
issue = task in our board